presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -172,6 +164,54 @@ else if (canQueueMore()) {
         }
     }
 
+    public ResourceGroupStateInfo getStateInfo()
+    {
+        synchronized (root) {
+            return new ResourceGroupStateInfo(
+                    id,
+                    getState(),
+                    new DataSize(softMemoryLimitBytes, BYTE),
+                    new DataSize(cachedMemoryUsageBytes, BYTE),
+                    maxRunningQueries,
+                    maxQueuedQueries,
+                    runningTimeLimit,
+                    queuedTimeLimit,
+                    getAggregatedRunningQueriesInfo(),
+                    queuedQueries.size() + descendantQueuedQueries);
```

Comment:
Yes, it's queuedQueries is empty for all internal nodes, and descendantQueuedQueries would be 0 for leaf.